components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java

```diff
@@ -907,6 +997,323 @@ public Response throttlingPoliciesCustomRuleIdDelete(String ruleId,  MessageCont
+            importPolicy = yamlMapper.readValue(fileReader, ExportThrottlePolicyDTO.class);
+        } else {
+            ObjectMapper mapper = new ObjectMapper();
+            JSONParser parser = new JSONParser();
+            Object obj;
+            obj = parser.parse(fileReader);
+            JSONObject jsonObject = (JSONObject) obj;
+            importPolicy = mapper.convertValue(jsonObject, ExportThrottlePolicyDTO.class);
+        }
+        return importPolicy;
+    }
+
+    /**
+     * Imports a Throttling policy with the overwriting capability
+     *
+     * @param fileInputStream Input stream from the REST request
+     * @param fileDetail      exportThrottlePolicyDTO Exported Throttling policy details
+     * @param overwrite       User can either update an existing throttling policy with the same name or let the conflict happen
+     * @return Response with message indicating the status of the importation and the imported/updated policy name
+     */
+    @Override
+    public Response importThrottlingPolicy(InputStream fileInputStream, Attachment fileDetail,
+            Boolean overwrite, MessageContext messageContext) throws APIManagementException {
+        ExportThrottlePolicyDTO exportThrottlePolicyDTO = null;
+        String policyType = "";
+        try {
+            exportThrottlePolicyDTO = getImportedPolicy(fileInputStream, fileDetail);
+        } catch (APIImportExportException | IOException | ParseException e) {
+            String errorMessage = "Error retrieving Throttling policy";
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        if (exportThrottlePolicyDTO != null) {
+            policyType = exportThrottlePolicyDTO.getSubtype();
+        } else {
+            String errorMessage = "Error resolving ExportThrottlePolicyDTO";
+            RestApiUtil.handleInternalServerError(errorMessage, log);
+        }
+        return resolveUpdateThrottlingPolicy(policyType, overwrite, exportThrottlePolicyDTO, messageContext);
+    }
+
+    /**
+     * @param policyType              Throttling policy type
+     * @param overwrite               User can either update an existing throttling policy with the same name or let the conflict happen
+     * @param exportThrottlePolicyDTO the policy to be imported
+     * @return Response with  message indicating the status of the importation and the imported/updated policy name
+     */
+    private Response resolveUpdateThrottlingPolicy(String policyType, boolean overwrite,
+            ExportThrottlePolicyDTO exportThrottlePolicyDTO, MessageContext messageContext)
+            throws APIManagementException {
+        ObjectMapper mapper = new ObjectMapper();
+        APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+        String username = RestApiCommonUtil.getLoggedInUsername();
+        if (policyType.equals(RestApiConstants.RESOURCE_SUBSCRIPTION_POLICY)) {
```

Comment:
swap the eqalling

Reply:
Fix in other places as well

Reply:
Fixed all  the places with 30d0ecb9c29c0024fa182ba597c8eda4c128c451